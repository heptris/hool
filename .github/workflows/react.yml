name: React CI

on:
  push:
    branches:
      - develop

jobs:
  build:
    runs-on: ubuntu-20.04

    steps:
      - uses: actions/checkout@v3

      - name: Create secrets
        run: echo "${{ secrets.FRONT_DOTENV_LOCAL }}" > frontend/.env.local

      - name: Set up Node.js 16.16
        uses: actions/setup-node@v3
        with:
          node-version: 16.16

      - name: Install dependencies
        run: |
          cd frontend/
          npm install

      # - name: Run tests
      #   run: |
      #     cd frontend/
      #     npm run test

      - name: Dockerize
        run: |
          echo "${{ secrets.FRONT_DOCKER_HUB_PW }}" | docker login -u "${{ secrets.FRONT_DOCKER_HUB_USER }}" --password-stdin
          docker build -t "${{ secrets.FRONT_IMAGE_NAME }}" ./frontend/
          docker push "${{ secrets.FRONT_IMAGE_NAME }}"
          docker logout
