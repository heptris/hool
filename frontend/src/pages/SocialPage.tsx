import { useState } from "react";

import styled from "styled-components";

import SocialHeader from "components/social/SocialHeader";
import MyFriends from "components/social/MyFriends";
import Requests from "components/social/Requests";

function SocialView() {
  const [isDisplayMyFriends, setIsDisplayMyFriends] = useState(true);

  const myFriends = [
    {
      profileImg: 0,
      nickname: "Thomas",
      email: "tho@gmail.com",
      curPos: "일본 vs 대한민국 같이봐요",
      status: "live",
    },
    {
      profileImg: 1,
      nickname: "Bell",
      email: "bell@gmail.com",
      curPos: "접속중",
      status: "loggedIn",
    },
    {
      profileImg: 2,
      nickname: "Dijkstra",
      email: "graph@gmail.com",
      curPos: "12 minutes ago",
      status: "loggedOut",
    },
    {
      profileImg: 3,
      nickname: "Perry",
      email: "graph@gmail.com",
      curPos: "12 minutes ago",
      status: "loggedOut",
    },
  ];

  const requests = [{ nickname: "Ronaldo", msg: "친해지고싶어요." }];

  const searchRes = [{ nickname: "Kruskal", email: "mst@gmail.com" }];

  return (
    <Row>
      <SocialHeader
        isDisplayMyFriends={isDisplayMyFriends}
        setIsDisplayMyFriends={setIsDisplayMyFriends}
      />
      {isDisplayMyFriends && <MyFriends myFriends={myFriends} />}
      {!isDisplayMyFriends && <Requests />}
    </Row>
  );
}

const Row = styled.div`
  width: 83.333333%;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: start;
  align-items: center;
  padding: 1rem 0 0 0;
`;

export default SocialView;
