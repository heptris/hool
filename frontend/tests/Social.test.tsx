import React from "react";
import { render } from "@testing-library/react";
import SocialPage from "../src/pages/SocialPage";

describe("<SocialPage />", () => {
  it("matches snapshot", () => {
    const utils = render(<SocialPage />);
    expect(utils.container).toMatchSnapshot();
  });

  it("shows the props correctly", () => {
    const utils = render(<SocialPage />);
    utils.getByText("친구찾기");
    utils.getByText("친구와 함께하면 즐거움이 두배에요.");
    utils.getByText("내친구");
    utils.getByText("친구요청");
  });
});
