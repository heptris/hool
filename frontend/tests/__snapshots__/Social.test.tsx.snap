// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SocialPage /> matches snapshot 1`] = `
<div>
  <div
    class="sc-bczRLJ dEIXcx"
  >
    <div
      class="sc-gsnTZi fIvucl"
    >
      <div
        style="display: flex; justify-content: space-between; align-items: center;"
      >
        <h1
          class="sc-dkzDqf ubOrv"
        >
          친구찾기
        </h1>
      </div>
      <span
        class="sc-hKMtZM fCQPYB"
      >
        친구와 함께하면 즐거움이 두배에요.
      </span>
      <div
        class="sc-eCYdqJ cAgjmv"
      >
        <div
          class="sc-gKXOVf"
        >
          <button
            class="sc-iBkjds kSKSbT"
          >
            <span
              style="color: rgb(0, 198, 193);"
            >
              내친구
            </span>
          </button>
          <button
            class="sc-iBkjds kSKSbT"
          >
            <span>
              친구요청
            </span>
          </button>
        </div>
      </div>
      <hr
        class="sc-jSMfEi bchyHn"
      />
    </div>
    <div
      class="sc-fEOsli iLpJxV"
    >
      <div
        class="sc-ftvSup ikFYvQ sc-papXJ fYSOyb"
      >
        <div
          class="sc-jqUVSM gLtmvn"
        >
          <img
            class="sc-kDDrLX ckljJQ"
            src="default"
          />
          <div
            class="sc-iqcoie cbRpQG"
          >
            <h1
              class="sc-crXcEl gtxtSW"
            >
              Thomas
            </h1>
            <h2
              class="sc-evZas kvSCWJ"
            >
              tho@gmail.com
            </h2>
            <span
              class="sc-breuTD hMqIoS"
            >
              일본 vs 대한민국 같이봐요
            </span>
          </div>
        </div>
        <div>
          <i
            class="sc-ksZaOG hbHzFs fa-solid fa-ellipsis-vertical"
          />
        </div>
      </div>
      <div
        class="sc-ftvSup ikFYvQ sc-papXJ fYSOyb"
      >
        <div
          class="sc-jqUVSM gLtmvn"
        >
          <img
            class="sc-kDDrLX ckljJQ"
            src="default"
          />
          <div
            class="sc-iqcoie cbRpQG"
          >
            <h1
              class="sc-crXcEl gtxtSW"
            >
              Bell
            </h1>
            <h2
              class="sc-evZas kvSCWJ"
            >
              bell@gmail.com
            </h2>
            <span
              class="sc-breuTD hMqIoS"
            >
              접속중
            </span>
          </div>
        </div>
        <div>
          <i
            class="sc-ksZaOG hbHzFs fa-solid fa-ellipsis-vertical"
          />
        </div>
      </div>
      <div
        class="sc-ftvSup ikFYvQ sc-papXJ fYSOyb"
      >
        <div
          class="sc-jqUVSM gLtmvn"
        >
          <img
            class="sc-kDDrLX ckljJQ"
            src="default"
          />
          <div
            class="sc-iqcoie cbRpQG"
          >
            <h1
              class="sc-crXcEl gtxtSW"
            >
              Dijkstra
            </h1>
            <h2
              class="sc-evZas kvSCWJ"
            >
              graph@gmail.com
            </h2>
            <span
              class="sc-breuTD hMqIoS"
            >
              12 minutes ago
            </span>
          </div>
        </div>
        <div>
          <i
            class="sc-ksZaOG hbHzFs fa-solid fa-ellipsis-vertical"
          />
        </div>
      </div>
      <div
        class="sc-ftvSup ikFYvQ sc-papXJ fYSOyb"
      >
        <div
          class="sc-jqUVSM gLtmvn"
        >
          <img
            class="sc-kDDrLX ckljJQ"
            src="default"
          />
          <div
            class="sc-iqcoie cbRpQG"
          >
            <h1
              class="sc-crXcEl gtxtSW"
            >
              Perry
            </h1>
            <h2
              class="sc-evZas kvSCWJ"
            >
              graph@gmail.com
            </h2>
            <span
              class="sc-breuTD hMqIoS"
            >
              12 minutes ago
            </span>
          </div>
        </div>
        <div>
          <i
            class="sc-ksZaOG hbHzFs fa-solid fa-ellipsis-vertical"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
